packages/ember-data/lib/serializers/json-serializer.js

```diff
@@ -1459,6 +1460,7 @@ var JSONSerializer = Serializer.extend({
     @return {Object} json The deserialized payload
   */
   extractSingle: function(store, typeClass, payload, id, requestType) {
+    Ember.deprecate('`serializer.normalizePayload` has been deprecated. Please use `serializer.normalizeResponse` with the new Serializer API to modify the payload.', this.normalizePayload === JSONSerializer.prototype.normalizePayload);
```

Comment:
This is causing thousands of deprecations to be thrown for my application that has a large payload. The logging of all the deprecations is slow and the application load time is impacted. Before this change application load time was nearly instant. Now it is almost 10 seconds.